spec/ParseAPI.spec.js

```diff
@@ -618,6 +618,36 @@ describe('miscellaneous', function() {
     });
   });
 
+  it('pointer reassign is working properly (#1288)', (done) => {
+    Parse.Cloud.beforeSave('GameScore', (req, res) => {
+
+      var obj = req.object;
+      if (obj.get('point')) {
+        return res.success();
+      }
+       var TestObject1 = Parse.Object.extend('TestObject1');
+       var newObj = new TestObject1({'key1': 1});
+
+       return newObj.save().then((newObj) => {
+         obj.set('point' , newObj);
+         res.success();
+       });
+    });
+    var pointId;
+    var obj = new Parse.Object('GameScore');
+    obj.set('foo', 'bar');
+    obj.save().then(() => {
+      expect(obj.get('point')).not.toBeUndefined();
+      pointId = obj.get('point').id;
+      expect(pointId).not.toBeUndefined();
+      obj.set('foo', 'baz');
+      return obj.save();
+    }).then((obj) => {
+      expect(obj.get('point').id).toEqual(pointId);
```

Comment:
Can you make sure to remove the hook when the test is done with it? Check @nlutsenko's tests which do this. Prevents pollution.